test/e2e/kustomizations.go

```diff
@@ -0,0 +1,169 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+)
+
+const (
+	initRetryPatch = `- op: add
+  path: /spec/template/spec/containers/1/env/-
+  value:
+    name: CONTROLLER_KONG_ADMIN_INIT_RETRY_DELAY
+    value: "%s"
+- op: add
+  path: /spec/template/spec/containers/1/env/-
+  value:
+    name: CONTROLLER_KONG_ADMIN_INIT_RETRIES
+    value: "%d"`
+
+	livenessProbePatch = `- op: replace
+  path: /spec/template/spec/containers/%[1]d/livenessProbe/initialDelaySeconds
+  value: %[2]d
+- op: replace
+  path: /spec/template/spec/containers/%[1]d/livenessProbe/timeoutSeconds
+  value: %[3]d
+- op: replace
+  path: /spec/template/spec/containers/%[1]d/livenessProbe/failureThreshold
+  value: %[4]d`
+)
+
+// getKustomizedManifest takes a base manifest Reader and a Kustomization, and returns the manifest after applying the
+// Kustomization. It overwrites the Kustomization's Bases; any existing Bases will be discarded.
+func getKustomizedManifest(baseManifestReader io.Reader, kustomization types.Kustomization) (io.Reader, error) {
+	workDir, err := os.MkdirTemp("", "kictest.")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(workDir)
+	orig, err := io.ReadAll(baseManifestReader)
+	if err != nil {
+		return nil, err
+	}
+	err = os.WriteFile(filepath.Join(workDir, "base.yaml"), orig, 0o600)
+	if err != nil {
+		return nil, err
+	}
+	kustomization.Bases = []string{"base.yaml"}
```

Comment:
As indicated earlier: if we can I'd go for using `Resources` here. Even the field itself has a `Deprecation` notice.

Better to address it now then at some point in the future when kustomize API will remove this completely.